mpcs/src/util/parallel.rs

```diff
@@ -1,7 +1,13 @@
 pub fn num_threads() -> usize {
```

Comment:
Giving another similar TODO issue https://github.com/scroll-tech/ceno/issues/861
I would suggest make this PR with slightly wider scope: remove this `parallel.rs` file and leverage plonky3 `maybe-rayon` to achieve the same functionality

Reply:
we will in favour of this solution instead https://github.com/scroll-tech/ceno/pull/897, for `parallel.rs` is some previous left-over and we will clean it up eventually